CONTRIBUTING.md

```diff
@@ -30,8 +30,12 @@ pip install --upgrade pip
 # Install dependencies
 pip install -r requirements.txt
 
-# Code formatting (Run this as a pre-commit step for your code changes)
+# Code formatting (run this as a pre-commit step for your code changes)
 black .
+
+# Code style and docstring style (run this as a pre-commit step for your code changes)
+ruff .
+
```

Comment:
we should point people to the make targets for all these things (installing deps, as well as code formatting).  That way if we change tools the docs don't need changing, we just fix the make targets.

Reply:
so you mean to have a Makefile target(s) to install `black` and `ruff`, right? Prob. with some checks that these tools are installed already...

Reply:
no, i mean we have a `make verify` target that runs black and ruff, so rather than telling people to explicitly run ruff and black, just tell them to run make verify (and make format)

Reply:
ah sure, will change it ASAP. Thanks

Reply:
aaand done